File: UI/Web/src/app/_models/series-detail/relation-kind.ts
export enum RelationKind {
    Prequel = 1,
    Sequel = 2,
    SpinOff = 3,
    Adaptation = 4,
    SideStory = 5,
    Character = 6,
    Contains = 7,
    Other = 8,
    AlternativeSetting = 9,
    AlternativeVersion = 10,
    Doujinshi = 11,
    /**
     * This is UI only. Backend will generate Parent series for everything but Prequel/Sequel
     */
    Parent = 12,
    Edition = 13,
    Annual = 14,

    Main = 15,
    Cameo = 16,
    CharacterFocus = 17,
    Compilation = 18,
    Crossover = 19,
    Expansion = 20,
    Parody = 21,
    Reboot = 22,
    Remake = 23,
    Series = 24,
    Source = 25,
    Summary = 26,
    Uncollected = 27

}

const RelationKindsUnsorted = [
    {text: 'Prequel', value: RelationKind.Prequel},
    {text: 'Sequel', value: RelationKind.Sequel},
    {text: 'Spin Off', value: RelationKind.SpinOff},
    {text: 'Adaptation', value: RelationKind.Adaptation},
    {text: 'Annual', value: RelationKind.Annual},
    {text: 'Alternative Setting', value: RelationKind.AlternativeSetting},
    {text: 'Alternative Version', value: RelationKind.AlternativeVersion},
    {text: 'Side Story', value: RelationKind.SideStory},
    {text: 'Character', value: RelationKind.Character},
    {text: 'Contains', value: RelationKind.Contains},
    {text: 'Edition', value: RelationKind.Edition},
    {text: 'Doujinshi', value: RelationKind.Doujinshi},
    {text: 'Other', value: RelationKind.Other},
    {text: 'Main Story', value: RelationKind.Main},
    {text: 'Cameo', value: RelationKind.Cameo},
    {text: 'Character Focus', value: RelationKind.CharacterFocus},
    {text: 'Compilation', value: RelationKind.Compilation},
    {text: 'Crossover', value: RelationKind.Crossover},
    {text: 'Expansion', value: RelationKind.Expansion},
    {text: 'Parody', value: RelationKind.Parody},
    {text: 'Reboot', value: RelationKind.Reboot},
    {text: 'Remake', value: RelationKind.Remake},
    {text: 'Series', value: RelationKind.Series},
    {text: 'Source', value: RelationKind.Source},
    {text: 'Summary', value: RelationKind.Summary},
    {text: 'Uncollected', value: RelationKind.Uncollected},
];

export const RelationKinds = RelationKindsUnsorted.slice().sort((a, b) => a.text.localeCompare(b.text));

File: Kavita.Models/Entities/Enums/RelationKind.cs
using System.ComponentModel;

namespace Kavita.Models.Entities.Enums;

/// <summary>
/// Represents a relationship between Series
/// </summary>
public enum RelationKind
{
    /// <summary>
    /// Story that occurred before the original.
    /// </summary>
    [Description("Prequel")]
    Prequel = 1,
    /// <summary>
    /// Direct continuation of the story.
    /// </summary>
    [Description("Sequel")]
    Sequel = 2,
    /// <summary>
    /// Uses characters of a different series, but is not an alternate setting or story.
    /// </summary>
    [Description("Spin Off")]
    SpinOff = 3,
    /// <summary>
    /// Manga/Anime/Light Novel adaptation
    /// </summary>
    [Description("Adaptation")]
    Adaptation = 4,
    /// <summary>
    /// Takes place sometime during the parent storyline.
    /// </summary>
    [Description("Side Story")]
    SideStory = 5,
    /// <summary>
    /// When characters appear in both series, but is not a spin-off
    /// </summary>
    [Description("Character")]
    Character = 6,
    /// <summary>
    /// When the story contains another story, useful for One-Shots
    /// </summary>
    [Description("Contains")]
    Contains = 7,
    /// <summary>
    /// When nothing else fits
    /// </summary>
    [Description("Other")]
    Other = 8,
    /// <summary>
    /// Same universe/world/reality/timeline, completely different characters
    /// </summary>
    [Description("Alternative Setting")]
    AlternativeSetting = 9,
    /// <summary>
    /// Same setting, same characters, story is told differently
    /// </summary>
    [Description("Alternative Version")]
    AlternativeVersion = 10,
    /// <summary>
    /// Doujinshi or Fan work
    /// </summary>
    [Description("Doujinshi")]
    Doujinshi = 11,
    /// <summary>
    /// This is a UI field only. Not to be used in backend
    /// </summary>
    [Description("Parent")]
    Parent = 12,
    /// <summary>
    /// Same story, could be translation, colorization... Different edition of the series
    /// </summary>
    [Description("Edition")]
    Edition = 13,
    /// <summary>
    /// The target series is an annual of the Series
    /// </summary>
    [Description("Annual")]
    Annual = 14,
    #region MangaBaka Only
    /// <summary>
    /// MangaBaka <c>main</c>: the parent/main story this entry branches from.
    /// </summary>
    [Description("Main Story")]
    Main = 15,
    /// <summary>
    /// MangaBaka <c>cameo</c>: a brief appearance of characters from another series.
    /// </summary>
    [Description("Cameo")]
    Cameo = 16,
    /// <summary>
    /// MangaBaka <c>character_focus</c>: focuses on characters from the related series.
    /// </summary>
    [Description("Character Focus")]
    CharacterFocus = 17,
    /// <summary>
    /// MangaBaka <c>compilation</c>: a compilation/collection of the related series.
    /// </summary>
    [Description("Compilation")]
    Compilation = 18,
    /// <summary>
    /// MangaBaka <c>crossover</c>: a crossover between series.
    /// </summary>
    [Description("Crossover")]
    Crossover = 19,
    /// <summary>
    /// MangaBaka <c>expansion</c>: an expanded version of the related series.
    /// </summary>
    [Description("Expansion")]
    Expansion = 20,
    /// <summary>
    /// MangaBaka <c>parody</c>: a parody of the related series.
    /// </summary>
    [Description("Parody")]
    Parody = 21,
    /// <summary>
    /// MangaBaka <c>reboot</c>: a reboot of the related series.
    /// </summary>
    [Description("Reboot")]
    Reboot = 22,
    /// <summary>
    /// MangaBaka <c>remake</c>: a remake of the related series.
    /// </summary>
    [Description("Remake")]
    Remake = 23,
    /// <summary>
    /// MangaBaka <c>series</c>: belongs to the same overall series.
    /// </summary>
    [Description("Series")]
    Series = 24,
    /// <summary>
    /// MangaBaka <c>source</c>: the source work this entry adapts.
    /// </summary>
    [Description("Source")]
    Source = 25,
    /// <summary>
    /// MangaBaka <c>summary</c>: a summary/abridged version of the related series.
    /// </summary>
    [Description("Summary")]
    Summary = 26,
    /// <summary>
    /// MangaBaka <c>uncollected</c>: uncollected chapters related to the series.
    /// </summary>
    [Description("Uncollected")]
    Uncollected = 27

    #endregion

}
